src/engine/SCons/Node/FS.py

```diff
@@ -3281,26 +3312,15 @@ def _build_dependency_map(self, binfo):
         # For an "empty" binfo properties like bsources
         # do not exist: check this to avoid exception.
         if (len(binfo.bsourcesigs) + len(binfo.bdependsigs) + \
-                len(binfo.bimplicitsigs)) == 0:
+            len(binfo.bimplicitsigs)) == 0:
             return {}
 
-        pairs = [
-            (binfo.bsources, binfo.bsourcesigs),
-            (binfo.bdepends, binfo.bdependsigs),
-            (binfo.bimplicit, binfo.bimplicitsigs)
-           ]
-
-        m = {}
-        for children, signatures in pairs:
-            for child, signature in zip(children, signatures):
-                schild = str(child)
-                m[schild] = signature
-
-
 
         # store this info so we can avoid regenerating it.
-        binfo.dependency_map = m
-        return m
+        binfo.dependency_map = { str(child):signature for child, signature in zip(chain(binfo.bsources, binfo.bdepends, binfo.bimplicit),
```

Comment:
Hmm, are the two nested calls of `chain` really needed here? Please check again.

Reply:
I only see one call to chain?  do you mean zip(chain(.. )?

Reply:
Yes, I mean the whole `zip()`call, so lines 3320 + 3321 together. The structure looks to me like `zip(chain(a, b, chain(c, d, e))`, can't this be reduced to `zip(chain(a, b, c, d, e))`?

Reply:
I think you're missing the str(child) part?

Reply:
Don't bother...I got confused over the "github diff" display in my browser, sorry. Line 3320 seems to end with `zip(chain(binfo.bsources, binfo.bdepends,` (if one doesn't scroll to the right, ahem) and combined with the l. 3321 `chain(binfo.bsourcesigs, binfo.bdependsigs, binfo.bimplicitsigs))` I didn't understand what you're trying to do here.

It's all good now, latch on! ;)